pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -121,9 +123,10 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, log *logrus.
 	// think is required. We expect payload to have everything else required.
 	case http.MethodPut:
 		ext = converter.ToExternal(&api.OpenShiftCluster{
-			ID:   doc.OpenShiftCluster.ID,
-			Name: doc.OpenShiftCluster.Name,
-			Type: doc.OpenShiftCluster.Type,
+			ID:         doc.OpenShiftCluster.ID,
+			Name:       doc.OpenShiftCluster.Name,
+			Type:       doc.OpenShiftCluster.Type,
+			SystemData: systemData,
```

Comment:
You may need a function which stomps systemData selectively.  I would call that function immediately after `doc.OpenShiftCluster.ID, doc.OpenShiftCluster.Name, doc.OpenShiftCluster.Type = oldID, oldName, oldType` - i.e. regardless of what the end user sends in PUT/PATCH, we will overwrite that data as we please